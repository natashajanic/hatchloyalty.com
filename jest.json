{
  "collectCoverage": true,
  "coveragePathIgnorePatterns": ["tests"],
  "globals": {},
  "moduleFileExtensions": [
    "ts",
    "tsx",
    "js"
  ],
  "moduleNameMapper": {
    "^.+\\.(css|less|scss)$": "identity-obj-proxy",
    "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/tests/assetsTransformer.ts",
    "^src/(.*)$": "<rootDir>/src/$1"
  },
  "setupFiles": [
    "raf/polyfill"
  ],
  "setupFilesAfterEnv": [
    "<rootDir>/tests/setupTests.ts"
  ],
  "snapshotSerializers": [
    "enzyme-to-json/serializer"
  ],
  "transform": {
    "\\.(ts|tsx)$": "ts-jest"
  },
  "testRegex": "/src/.*/__tests__/.*\\.(ts|tsx|js)$",
  "unmockedModulePathPatterns": [],
  "verbose": true
}
