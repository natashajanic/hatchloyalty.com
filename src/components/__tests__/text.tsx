import { mount, ReactWrapper } from 'enzyme'
import 'jest' // tslint:disable-line no-import-side-effect
import * as React from 'react'
import Text from '../text'

describe('Text', () => {
  let mountedComponent: ReactWrapper<any, any, any>

  beforeEach(() => {
    mountedComponent = mount(<Text>Test 123</Text>)
  })

  describe('render', () => {
    it('matches the expected snapshot', async () => {
      expect(mountedComponent).toMatchSnapshot()
    })

    it('renders properly setup to truncate', async () => {
      mountedComponent.setProps({ truncate: 3 })
      expect(mountedComponent).toMatchSnapshot()
    })
  })

  it('has a display name', () => {
    expect(Text.displayName).toEqual('Text')
  })
})
