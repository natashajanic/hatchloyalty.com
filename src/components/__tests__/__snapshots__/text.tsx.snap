// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text render matches the expected snapshot 1`] = `
<Text
  blacklist={
    Array [
      "css",
      "fontSize",
      "lineHeight",
      "m",
      "mt",
      "mr",
      "mb",
      "ml",
      "mx",
      "my",
      "p",
      "pt",
      "pr",
      "pb",
      "pl",
      "px",
      "py",
      "textStyle",
      "alignItems",
      "color",
      "bg",
      "display",
      "flex",
      "fontWeight",
      "textAlign",
    ]
  }
  fontSize={3}
  lineHeight="1.5"
  m={0}
  textStyle="normal"
>
  <StyledComponent
    blacklist={
      Array [
        "css",
        "fontSize",
        "lineHeight",
        "m",
        "mt",
        "mr",
        "mb",
        "ml",
        "mx",
        "my",
        "p",
        "pt",
        "pr",
        "pb",
        "pl",
        "px",
        "py",
        "textStyle",
        "alignItems",
        "color",
        "bg",
        "display",
        "flex",
        "fontWeight",
        "textAlign",
      ]
    }
    fontSize={3}
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "lastClassName": "gytoeM",
          "rules": Array [
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
          ],
        },
        "defaultProps": Object {
          "blacklist": Array [
            "css",
            "fontSize",
            "lineHeight",
            "m",
            "mt",
            "mr",
            "mb",
            "ml",
            "mx",
            "my",
            "p",
            "pt",
            "pr",
            "pb",
            "pl",
            "px",
            "py",
            "textStyle",
            "alignItems",
            "color",
            "bg",
            "display",
            "flex",
            "fontWeight",
            "textAlign",
          ],
          "fontSize": 3,
          "lineHeight": "1.5",
          "m": 0,
          "textStyle": "normal",
        },
        "displayName": "Text",
        "foldedComponentIds": Array [],
        "propTypes": Object {
          "alignItems": [Function],
          "bg": [Function],
          "color": [Function],
          "display": [Function],
          "flex": [Function],
          "fontSize": [Function],
          "fontWeight": [Function],
          "lineHeight": [Function],
          "m": [Function],
          "mb": [Function],
          "ml": [Function],
          "mr": [Function],
          "mt": [Function],
          "mx": [Function],
          "my": [Function],
          "p": [Function],
          "pb": [Function],
          "pl": [Function],
          "pr": [Function],
          "pt": [Function],
          "px": [Function],
          "py": [Function],
          "textAlign": [Function],
          "textStyle": [Function],
        },
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "systemComponent": true,
        "target": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    lineHeight="1.5"
    m={0}
    textStyle="normal"
  >
    <ForwardRef
      blacklist={
        Array [
          "css",
          "fontSize",
          "lineHeight",
          "m",
          "mt",
          "mr",
          "mb",
          "ml",
          "mx",
          "my",
          "p",
          "pt",
          "pr",
          "pb",
          "pl",
          "px",
          "py",
          "textStyle",
          "alignItems",
          "color",
          "bg",
          "display",
          "flex",
          "fontWeight",
          "textAlign",
        ]
      }
      className="sc-bdVaJa gytoeM"
      fontSize={3}
      lineHeight="1.5"
      m={0}
      textStyle="normal"
    >
      <div
        className="sc-bdVaJa gytoeM"
      >
        Test 123
      </div>
    </ForwardRef>
  </StyledComponent>
</Text>
`;

exports[`Text render renders properly setup to truncate 1`] = `
<Text
  blacklist={
    Array [
      "css",
      "fontSize",
      "lineHeight",
      "m",
      "mt",
      "mr",
      "mb",
      "ml",
      "mx",
      "my",
      "p",
      "pt",
      "pr",
      "pb",
      "pl",
      "px",
      "py",
      "textStyle",
      "alignItems",
      "color",
      "bg",
      "display",
      "flex",
      "fontWeight",
      "textAlign",
    ]
  }
  fontSize={3}
  lineHeight="1.5"
  m={0}
  textStyle="normal"
  truncate={3}
>
  <StyledComponent
    blacklist={
      Array [
        "css",
        "fontSize",
        "lineHeight",
        "m",
        "mt",
        "mr",
        "mb",
        "ml",
        "mx",
        "my",
        "p",
        "pt",
        "pr",
        "pb",
        "pl",
        "px",
        "py",
        "textStyle",
        "alignItems",
        "color",
        "bg",
        "display",
        "flex",
        "fontWeight",
        "textAlign",
      ]
    }
    fontSize={3}
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "lastClassName": "iFjSNO",
          "rules": Array [
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
          ],
        },
        "defaultProps": Object {
          "blacklist": Array [
            "css",
            "fontSize",
            "lineHeight",
            "m",
            "mt",
            "mr",
            "mb",
            "ml",
            "mx",
            "my",
            "p",
            "pt",
            "pr",
            "pb",
            "pl",
            "px",
            "py",
            "textStyle",
            "alignItems",
            "color",
            "bg",
            "display",
            "flex",
            "fontWeight",
            "textAlign",
          ],
          "fontSize": 3,
          "lineHeight": "1.5",
          "m": 0,
          "textStyle": "normal",
        },
        "displayName": "Text",
        "foldedComponentIds": Array [],
        "propTypes": Object {
          "alignItems": [Function],
          "bg": [Function],
          "color": [Function],
          "display": [Function],
          "flex": [Function],
          "fontSize": [Function],
          "fontWeight": [Function],
          "lineHeight": [Function],
          "m": [Function],
          "mb": [Function],
          "ml": [Function],
          "mr": [Function],
          "mt": [Function],
          "mx": [Function],
          "my": [Function],
          "p": [Function],
          "pb": [Function],
          "pl": [Function],
          "pr": [Function],
          "pt": [Function],
          "px": [Function],
          "py": [Function],
          "textAlign": [Function],
          "textStyle": [Function],
        },
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "systemComponent": true,
        "target": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    lineHeight="1.5"
    m={0}
    textStyle="normal"
    truncate={3}
  >
    <ForwardRef
      blacklist={
        Array [
          "css",
          "fontSize",
          "lineHeight",
          "m",
          "mt",
          "mr",
          "mb",
          "ml",
          "mx",
          "my",
          "p",
          "pt",
          "pr",
          "pb",
          "pl",
          "px",
          "py",
          "textStyle",
          "alignItems",
          "color",
          "bg",
          "display",
          "flex",
          "fontWeight",
          "textAlign",
        ]
      }
      className="sc-bdVaJa iFjSNO"
      fontSize={3}
      lineHeight="1.5"
      m={0}
      textStyle="normal"
      truncate={3}
    >
      <div
        className="sc-bdVaJa iFjSNO"
        truncate={3}
      >
        Test 123
      </div>
    </ForwardRef>
  </StyledComponent>
</Text>
`;
